kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -340,39 +337,16 @@ public void addSharedInformerEventListener(SharedInformerEventListener event) {
     this.eventListeners.add(event);
   }
 
-  Map<String, SharedIndexInformer> getInformers() {
-    return this.informers;
-  }
-
-  static String getInformerKey(OperationContext operationContext) {
-    StringBuilder keyBuilder = new StringBuilder();
-    if (operationContext.getApiGroupName() == null) {
-      keyBuilder.append(operationContext.getApiGroupVersion());
-    } else {
-      keyBuilder.append(operationContext.getApiGroupName()).append("/").append(operationContext.getApiGroupVersion());
+  private synchronized void startInformer(SharedIndexInformer informer) {
+    try {
+      informer.run();
```

Comment:
With other informer changes, run is a sync call.  It probably makes sense for the factory to start async though.  However if you think sync is better then that would remove the need for the listeners.